pkg/controller/unifiedpushserver/postgresql.go

```diff
@@ -34,21 +35,35 @@ func newPostgresqlPersistentVolumeClaim(cr *pushv1alpha1.UnifiedPushServer) (*co
 }
 
 func newPostgresqlSecret(cr *pushv1alpha1.UnifiedPushServer) (*corev1.Secret, error) {
-	databasePassword, err := generatePassword()
-	if err != nil {
-		return nil, err
-	}
+	externaldb := &cr.Spec.ExternalDB
+	if !*externaldb {
```

Comment:
```suggestion
	if !cr.Spec.ExternalDB {
```

- Doesn't need to be a pointer (at least while this is a required field)
- Doesn't need to be an explicit local variable when this is the only place it's called